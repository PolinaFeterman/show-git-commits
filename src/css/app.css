.app {
  text-align: center;
}
.header {
  height: 100px;
  background-color: #edf7fc;
  margin-bottom: 20px;
}
