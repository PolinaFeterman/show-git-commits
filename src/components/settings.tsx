import "../css/settings.css";
import AppContextInterface from "../interfaces/app-context-interface";
import { AppCtx } from "../app";
import React,{ FormEventHandler, useContext, useEffect, useState } from "react";
import { useNavigate } from "react-router-dom";

export const Settings = () => {
  const appContext = React.useContext(AppCtx) as AppContextInterface;

  const [org, setOrg] = React.useState("");
  const [repo, setRepo] = React.useState("");
  const [perPage, setPerPage] = React.useState(appContext.perPage.toString());

  const navigate = useNavigate();
  const submitForm = (event: React.FormEvent<HTMLFormElement>) => {
    appContext.perPage = Number(perPage);
    event.preventDefault();
    return navigate(`/${org}/${repo}`);
  };

  return (
    <form onSubmit={submitForm} data-testid="form">
      <div className="input">
        <label id="organization" className="input label">Organization:</label>
        <input
          className="rounded"
          aria-labelledby="organization"
          name="organization"
          type="text"
          value={org}
          onChange={(e) => setOrg(e.target.value)}
        />
      </div>
      <div className="input">
        <label id="repo" className="label">Repo:</label>
        <input
          className="rounded"
          aria-labelledby="repo"
          name="repo"
          type="text"
          value={repo}
          onChange={(e) => setRepo(e.target.value)}
        />
      </div>
      <div className="input">
        <label id="perPage" className="label">Commits per page:</label>
        <input
          className="rounded"
          aria-labelledby="perPage"
          name="perPage"
          type="number"
          value={perPage}
          onChange={(e) => setPerPage(e.target.value)}
        />
      </div>
      <div className="button">
        <button type="submit" className="rounded" disabled={!repo || !org}>
          Submit
        </button>
      </div>
    </form>
  );
};
