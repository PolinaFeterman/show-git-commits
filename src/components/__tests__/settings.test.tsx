import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import React from "react";
import { BrowserRouter, Router, useNavigate } from "react-router-dom";
import renderer from "react-test-renderer";
import { AppCtx } from "../../app";
import { createMemoryHistory } from "history";

import { Settings } from "../settings";

describe("Settings", () => {
  test("should render elements", () => {
    render(
      <AppCtx.Provider
        value={{
          perPage: 10,
        }}
      >
        <BrowserRouter>
          <Settings />
        </BrowserRouter>
      </AppCtx.Provider>
    );
    expect(screen.getByText("Organization:")).toBeInTheDocument();
    expect(screen.getByText("Repo:")).toBeInTheDocument();
    expect(screen.getByText("Commits per page:")).toBeInTheDocument();
    expect(screen.getByText("Submit")).toBeInTheDocument();
  });
  test("should render correctly", () => {
    const tree = renderer
      .create(
        <AppCtx.Provider
          value={{
            perPage: 10,
          }}
        >
          <BrowserRouter>
            <Settings />
          </BrowserRouter>
        </AppCtx.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
//   test("should click the button", async () => {
//     const mockedNavigate = jest.fn();
//     jest.mock('react-router-dom', () => ({
//         ...jest.requireActual('react-router-dom') as any,
//        useNavigate: () => mockedNavigate,
//      }));
//     render(
//       <AppCtx.Provider
//         value={{
//           perPage: 10,
//         }}
//       >
//         <BrowserRouter>
//           <Settings />
//         </BrowserRouter>
//       </AppCtx.Provider>
//     );
//     const organization = screen.getByLabelText(
//       "Organization:"
//     ) as HTMLInputElement;
//     fireEvent.change(organization, {
//       target: { value: "m3db" },
//     });

//     expect(organization.value).toBe("m3db");

//     const repo = screen.getByLabelText("Repo:") as HTMLInputElement;
//     fireEvent.change(repo, {
//       target: { value: "m3" },
//     });

//     expect(repo.value).toBe("m3");

//     const submitButton = screen.getByRole("button");

//     fireEvent.click(submitButton);
//     expect(mockedNavigate).toHaveBeenCalledTimes(1);
//   });
  //   test("should submit correctly", async () => {
  //     const tree = renderer.create(
  //       <AppCtx.Provider
  //         value={{
  //           perPage: 10,
  //         }}
  //       >
  //         <BrowserRouter>

  //           <Settings />
  //         </BrowserRouter>
  //       </AppCtx.Provider>
  //     );
  //     expect(tree).toMatchSnapshot();
  //     const instance= tree.getInstance();
  //     const submitButton = await tree.root.findByProps({
  //       type: "submit",
  //     });
  //     expect(submitButton).toBeDefined();
  //     const organization = await tree.root.findByProps({
  //       name: "organization",
  //     });
  //     expect(organization).toBeDefined();
  //     const repo = await tree.root.findByProps({
  //         name: "repo",
  //       });
  //       expect(repo).toBeDefined();
  //       organization.props.value="m3db";
  //       repo.props.value="m3";
  //       expect(tree).toMatchSnapshot();
  //   });
});
