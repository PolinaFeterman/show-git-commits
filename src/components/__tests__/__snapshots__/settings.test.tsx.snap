// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings should render correctly 1`] = `
<form
  data-testid="form"
  onSubmit={[Function]}
>
  <div
    className="input"
  >
    <label
      className="input label"
      id="organization"
    >
      Organization:
    </label>
    <input
      aria-labelledby="organization"
      className="rounded"
      name="organization"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div
    className="input"
  >
    <label
      className="label"
      id="repo"
    >
      Repo:
    </label>
    <input
      aria-labelledby="repo"
      className="rounded"
      name="repo"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div
    className="input"
  >
    <label
      className="label"
      id="perPage"
    >
      Commits per page:
    </label>
    <input
      aria-labelledby="perPage"
      className="rounded"
      name="perPage"
      onChange={[Function]}
      type="number"
      value="10"
    />
  </div>
  <div
    className="button"
  >
    <button
      className="rounded"
      disabled={true}
      type="submit"
    >
      Submit
    </button>
  </div>
</form>
`;
